### PR TITLE
[python/asyncio] explicitly close client session and asynchronous context manager

### PR DESCRIPTION
**Pull request by tomplus**
From https://github.com/OpenAPITools/openapi-generator/pull/5094/ python clients expose api-client.close() method and the context manager. In this PR I added closing aiohttp.ClientSession there to allow users clean up objects in a more predictive way.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc: @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @Jyhess (2019/01) @slash-arun (2019/11) @spacether (2019/11)

Thank you reviewing and merging.

### CODE REVIEW

File: samples/client/petstore/python-asyncio/tests/test_pet_api.py [hunk 1]
```diff
@@ -27,15 +27,6 @@
 HOST = 'http://localhost:80/v2'
 
 
-def async_test(f):
-    def wrapper(*args, **kwargs):
-        coro = asyncio.coroutine(f)
-        future = coro(*args, **kwargs)
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(future)
-    return wrapper
```
Comment (spacether) [inline samples/client/petstore/python-asyncio/tests/test_pet_api.py]: I only see this being used in this one file. Why did you move it to the util file?
If it is used in more than one file it makes sense to put it there as a common utility tool, but if it is only needed here can you keep it here?

File: samples/client/petstore/python-asyncio/petstore_api/rest.py [hunk 2]
```diff
@@ -85,8 +85,9 @@ def __init__(self, configuration, pools_size=4, maxsize=None):
                 connector=connector
             )
 
-    def __del__(self):
-        asyncio.ensure_future(self.pool_manager.close())
+    def close(self):
+        if not self.pool_manager.closed:
+            asyncio.ensure_future(self.pool_manager.close())
```
Comment (spacether) [inline samples/client/petstore/python-asyncio/petstore_api/rest.py]: Can you add a test that checks the pool manager and ensures that it is closed like we did here:
samples/client/petstore/python-experimental/tests/test_api_client.py?

File: samples/client/petstore/python-asyncio/tests/test_pet_api.py (hunk 1 above)
Comment (tomplus) [reply samples/client/petstore/python-asyncio/tests/test_pet_api.py]: Yeah, my mistake... I've just added new file.

File: samples/client/petstore/python-asyncio/petstore_api/rest.py (hunk 2 above)
Comment (tomplus) [reply samples/client/petstore/python-asyncio/petstore_api/rest.py]: There is such test. Sorry, I forgot to commit it before.

File: samples/client/petstore/python-asyncio/tests/test_pet_api.py (hunk 1 above)
Comment (spacether) [reply samples/client/petstore/python-asyncio/tests/test_pet_api.py]: Looks good as-is

File: samples/client/petstore/python-asyncio/tests/test_api_client.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import os
+import unittest
+import asyncio
+import weakref
+
+from tests.util import async_test
+import petstore_api
+
+HOST = 'http://localhost:80/v2'
```
Comment (spacether) [inline samples/client/petstore/python-asyncio/tests/test_api_client.py]: Can you delete HOST, I don't think we use it here

File: samples/client/petstore/python-asyncio/petstore_api/rest.py (hunk 2 above)
Comment (spacether) [reply samples/client/petstore/python-asyncio/petstore_api/rest.py]: Test is below

File: samples/client/petstore/python-asyncio/tests/test_api_client.py (hunk 3 above)
Comment (tomplus) [reply samples/client/petstore/python-asyncio/tests/test_api_client.py]: Sure, removed.

File: samples/client/petstore/python-asyncio/tests/test_api_client.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            # thread pool
+            self.assertIsNotNone(client.pool)
+            pool_ref = weakref.ref(client._pool)
+            self.assertIsNotNone(pool_ref())
+            # pool_manager
+            self.assertFalse(client.rest_client.pool_manager.closed)
+            rest_pool_ref = client.rest_client.pool_manager
+
+        self.assertIsNone(pool_ref())
+        self.assertFalse(rest_pool_ref.closed)
```
Comment (spacether) [inline samples/client/petstore/python-asyncio/tests/test_api_client.py]: When do we check that the pool_manager is closed?

Comment (tomplus) [reply samples/client/petstore/python-asyncio/tests/test_api_client.py]: Thanks for spotting this. I had to switch to the asynchronous context generator to make it works. Please take a look again.

**Comment by spacether** [comment]
Closing and re-opening this to kick off CI tests again. It looks like the shippable error is unrelated to your code.

**Comment by wing328** [comment]
If no further feedback or comment on this PR, I'll merge it tomorrow (Thur).

**Comment by wing328** [comment]
@tomplus thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952
